Rebalances OT casings

GoldenDarkness55 opened the pull request:
# About the pull request

Was going to touch up the RPG since the two existing attempts aren't too good (#7070 #7072), but figured I might aswell fix up the rest of it into a less cancerous state, since for some reason no one's done it yet despite years of complaints.

Mostly reducing the explosive radius to curb braindead screenwide death, while making the other two underused properties better (fire and shrapnel). Should make the OT less ridiculous to play against and hopefully introduce alternatives to spamming the same full explosive rocket/mortar recipes round after round. Rounded up some numbers cause it bothers me if it doesn't end in a 0, easier to memorize that way too.
Played OT in the past, but didn't bother testing the new values so explosive falloff might require further tweaks to either side for desired intent.
I'll write down the proper numbers into the CL if this gets accepted, can't be arsed otherwise.

# Explain why it's good for the game

- RPG: increased falloff, reduced flame radius and explosive power, buffed fire intensity and shrapnel amount. Custom rocket range -1 tile to 7 and raised accuracy to tier 5

Instant screenwide boom/fire bad. Closer to AP but not completely without AoE.

- Mortar: Nerfed explosive radius, buffed fire intensity and shrapnel amount

Last I checked this wasn't a HE OB shell, should be a little bit saner.

- Small grenades: Raised container volume to 120 and slightly metal cost to compensate. Rounded numbers, raised fire intensity and shrapnel a bit.

90 is way too little.

- Large grenades: Nerfed explosive radius, buffed fire intensity, fire radius, duration and shard amount

Similar to RPG, slightly nerfing the screenwide explosions while making the other properties better

- C4: nerfed fire and explosive radius, fire duration, buffed damage to adjacent walls, explosive power, shrapnel amount and doubled fire intensity

Felt weird for c4 to have such large AoE, more power in a smaller radius makes more sense.

- Claymore: nerfed flame radius and explosive radius, buffed, fire intensity and shard amount

Less AoE to make it safer to use.

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>


# Changelog
:cl:
balance: OT M40 grenade casing volume from 90 to 120, power 175 to 180, falloff 75 to 80, shards 32 to 40, fire intensity 20 to 25, metal cost from 3750 to 4250 which is 12% of a single metal sheet
balance: OT M15 grenade casing power 215 to 220, falloff 90 to 120, shards 32 to 80, fire radius from 5 to 6, fire intensity 20 to 30 and fire duration 24 to 32
balance: OT claymore casing power 105 to 100, falloff 60 to 80, shards 32 to 40, fire radius 5 to 4, fire intensity 12 to 20
balance: OT C4 casing damage multiplier against walls from 1.5 to 2, power 260 to 280, falloff 90 to 120, shards 64 to 100, fire radius 6 to 4, fire intensity 26 to 50, duration 30 to 20
balance: OT rocket casing power 240 to 220, falloff 90 to 160, shards 64 to 80, fire radius 6 to 4, fire intensity 40 to 45
balance: OT mortar casing falloff 90 to 130, shards 128 to 200, fire intensity 40 to 45
balance: Custom rocket accuracy raised from tier 2 to 5 and range lowered from 8 to 7
/:cl:

TheManWithNoHands commented:
Only one note really.

Shrapnel buff doesnt do anything really against xenos. There is a reason why HEFA is never baugth unless for a joke or HvH (suprisingly i dont see it baugth it HvH as often as it should.) Shrapnel is not really that dmaging to xenos, mostly marines.
The shrapnel increase is honestly more so a nerf for marines.

Xenos dont have to worry about internal organ dmg after all.

GoldenDarkness55 commented:
> Only one note really.
> 
> Shrapnel buff doesnt do anything really against xenos. There is a reason why HEFA is never baugth unless for a joke or HvH (suprisingly i dont see it baugth it HvH as often as it should.) Shrapnel is not really that dmaging to xenos, mostly marines. The shrapnel increase is honestly more so a nerf for marines.
> 
> Xenos dont have to worry about internal organ dmg after all.

I'm aware. People can look into buffing dmg per shrap or adding more features (like phoron turning it incendiary) to it via special materials - the dmg taken bonus (hornet airburst shrapnel applies it iirc) seems like a decent idea. Im just putting the base possibility to mess with it out there, since shrap cap was unreasonably low and forgotten.

TheManWithNoHands commented:
> I'm aware. People can look into buffing dmg per shrap or adding more features (like phoron turning it incendiary) to it via special materials - the dmg taken bonus (hornet airburst shrapnel applies it iirc) seems like a decent idea. Im just putting the base possibility to mess with it out there, since shrap cap was unreasonably low and forgotten.

Thats not a reason to change something. Ever. Especially not when its just a simple number change.
I get your reasoning, and its nice, but you dont make changes for "someone else to finish".
It doesnt work, ever. Because your making changes for another PR that doesnt exsit.

If you change something, always make it self contained. The shrapnel increase, is for a completly diffrent PR. Not this one.
And i assume you arent interested in making said PR. So dont go and start the first changes for it.

And just to make this clear, this isnt about how bad it would be of a nerf or something. Its only a minor thing.
Its about the aditude behind the change. Dont change stuff with the reasoning of "someone else can do more to it later".
No. They wont. You dont make changes because someone MIGTH do whats needed to give them reason.
This mentallity can cause serious problems down the line. Its best you get rid of it. 'cause i member you having that reasoning before. And it doesnt work. EVER.

If your PR doesnt justify your changes itself, then it doesnt need the change.
And the shrapnel increase, isnt justified.

GoldenDarkness55 commented:
> > I'm aware. People can look into buffing dmg per shrap or adding more features (like phoron turning it incendiary) to it via special materials - the dmg taken bonus (hornet airburst shrapnel applies it iirc) seems like a decent idea. Im just putting the base possibility to mess with it out there, since shrap cap was unreasonably low and forgotten.
> 
> Thats not a reason to change something. Ever. Especially not when its just a simple number change. I get your reasoning, and its nice, but you dont make changes for "someone else to finish". It doesnt work, ever. Because your making changes for another PR that doesnt exsit.
> 
> If you change something, always make it self contained. The shrapnel increase, is for a completly diffrent PR. Not this one. And i assume you arent interested in making said PR. So dont go and start the first changes for it.
> 
> And just to make this clear, this isnt about how bad it would be of a nerf or something. Its only a minor thing. Its about the aditude behind the change. Dont change stuff with the reasoning of "someone else can do more to it later". No. They wont. You dont make changes because someone MIGTH do whats needed to give them reason. This mentallity can cause serious problems down the line. Its best you get rid of it. 'cause i member you having that reasoning before. And it doesnt work. EVER.
> 
> If your PR doesnt justify your changes itself, then it doesnt need the change. And the shrapnel increase, isnt justified.
> 
> I know im a bit repeating myself. But i really want to make sure you understand the problem.

The shrapnel increase is self contained. Shrapnel is just weak codewise right now, that's all there is to it. Someone buffing shrapnel in a separate PR is something entirely separate from rebalancing containers. funny textwall tho

TheManWithNoHands commented:
> The shrapnel increase is self contained. Shrapnel is just weak codewise right now, that's all there is to it. Someone buffing shrapnel in a separate PR is something entirely separate from rebalancing containers.
> 
>  I.e im not adding anything new that needs to actually be balanced, im just finetuning container numbers. What you're doing is the equivalent of whining someone didn't touch ammo values when adjusting magazine or ammo box sizes. funny textwall tho

My friend. I am putting a textwall there, because you cannot understand.
Because you are implementing a nerf, trying to use it as a buff, that needs future work to be a buff.

The shrapnel increase is a NERF.
It does not work for what your trying to acomplish. And the fact that future PR's can expand on it, doesnt justify it.

Its genuinely advise, and not whining. Because your fucking up a bit with the balance changes and made a nerf when trying to buff something.

GoldenDarkness55 commented:
> > The shrapnel increase is self contained. Shrapnel is just weak codewise right now, that's all there is to it. Someone buffing shrapnel in a separate PR is something entirely separate from rebalancing containers.
> > I.e im not adding anything new that needs to actually be balanced, im just finetuning container numbers. What you're doing is the equivalent of whining someone didn't touch ammo values when adjusting magazine or ammo box sizes. funny textwall tho
> 
> My friend. I am putting a textwall there, because you cannot understand. Because you are implementing a nerf, trying to use it as a buff, that needs future work to be a buff. I told you that in my first comment, and you excused it with "someone esle can make stuff with shrapnel. Im just letting them mess around with it"
> 
> The shrapnel increase is a NERF. It does not work for what your trying to acomplish. And the fact that future PR's can expand on it, doesnt justify it. You almost TRIPPELD the shrapnel ammount on the M15. An explosives main dmg source is the explosion or fire. The shrapnel is only for HvH combat really. What you did, is make the grenade require more skill to dmg xenos, and less skill to dmg marines. You keept the risk of using it high, but the reward is no longer equivelant.
> 
> Its genuinely advise, and not whining. Because your fucking up a bit with the balance changes and made a nerf when trying to buff something. Revert the shrapnel changes. It doesnt work.

Its not meant to be a shrapnel buff. It's a shrapnel capacity buff, shrapnel remains as it is.
I increased the amount to let people experiment with it more since it was too constrained. I couldn't care less about shrapnel still being useless or not, not part of the scope here. If someone wants to tackle shrapnel's issues that they can do it in their own PRs on their own time. You can already make shitty FF bombs and raising shrapnel caps isn't going to impact that in any way, the limit of 32 or 64 is equally as annoying and damaging as the higher ones.

i.e im not buffing shrapnel as a projectile in this PR and im certainly not buffing OT as a whole. I don't think I can possibly be any more clear about this.

RenaRenaRe commented:
So I hate to be the one to contest an OT buff but I feel like this needs to be said. I don't think the falloff changes have the effect you think they will have.
Let's use M15 as an example. The classic recipe people use for this is 60 cyclo 120 anfo. This gives 210 power and -96 falloff. Not technically maxed power but close enough and the numbers are nice (60 is one bottle) so it's a popular recipe. The thing to keep in mind is that falloff has a minimum of 25, it will not go lower than that. Given the base falloff of 90 the -96 falloff will easily get you to that minimum, so it's not even something you have to think about. You could tweak the recipe a bit to have more cyclo and less anfo and it would still be minimum falloff.

Now if we look at your changes, the base falloff has been nerfed to 110. But the thing is, 110-96 still easily hits the minimum of 25. So for full HE nades the falloff nerf will actually not have any effect at all. This would only affect hybrid nades that aren't maxing out on explosives, but those are not very common because they're already very suboptimal.
The same can be said for mortars. Standard recipe is 240 cyclo which also happens to give -96 falloff, enough to hit the 25 minimum even with the falloff nerf in this PR.

Rockets, C4 and I guess m40s I think are the only things that are meaningfully affected by the falloff nerfs here. By the way you mention C4 is supposed to have "more power in a smaller radius" but since falloff is applied for every tile the explosion spreads, that +20 max power is negated after only 1 tile of spread. So really with this C4 change it is only more powerful at point blank, any other distance and it's weaker. It's a pretty heavy nerf to an already underused casing. For rocket it makes sense since you can direct hit with it anyway.

(also I'm sad about claymore nerf but it is maybe deserved given current ones can insta kill runners and lurkers if you build them right)

iloveloopers commented:
Basically what Rena said though I don't know what possible claymore mix you could use to instakill runners as they can just run past the explosion (custom claymores don't trigger instantly like normal ones) and lurkers are too tanky.

Custom C4s have a 3 second warning telegraph, they aren't used to break down walls and are mostly for last resort situations. Normal C4 is way more cost effective for actual demolition.

Just rework the rockets to have some delay or something, thats what the main complaint with OT is, all other casings have counterplay.

GoldenDarkness55 commented:
> Now if we look at your changes, the base falloff has been nerfed to 110. But the thing is, 110-96 still easily hits the minimum of 25. So for full HE nades the falloff nerf will actually not have any effect at all. This would only affect hybrid nades that aren't maxing out on explosives, but those are not very common because they're already very suboptimal. The same can be said for mortars. Standard recipe is 240 cyclo which also happens to give -96 falloff, enough to hit the 25 minimum even with the falloff nerf in this PR.

I buffed the fire intensity max to make it more appealing as an alternative - if I increase the falloff any further to combat minmaxing for HE maxcap it'll start affecting all combinations too much. If people start only making these HE maxcaps and they'll have the same unfair effect it'll have to get nerfed like the rocket but I dont wanna take that route with it yet. Fair point about the mortar though, was meant to be 130.

>By the way you mention C4 is supposed to have "more power in a smaller radius" but since falloff is applied for every tile the explosion spreads, that +20 max power is negated after only 1 tile of spread. So really with this C4 change it is only more powerful at point blank, any other distance and it's weaker. It's a pretty heavy nerf to an already underused casing

I meant mostly the fire and wall damage changes, less AoE total on c4 was intended. For it to stop being "underused" it'd have to be incredibly OP as c4 isn't the most common tool around, I trust it might be used more just because of knocking the rocket meta.

RenaRenaRe commented:
Honestly I'm not complaining about the falloff numbers, I'll take OT buffs any day, I just thought it might be unintended for it to not have any real effect on standard HE recipes. I just wanted to point it out but if you're aware of it and it's intended then great.

ihatethisengine commented:
Shrapnel is not as weak as people think. Look at breaching charge for example, that can legit one shot a T3 behind a wall. Well placed air burst nade is also quite damaging. Main problem with shrapnel is that explosions tend to stun xenos, which makes shrapnel miss them completely.

GoldenDarkness55 commented:
> Shrapnel is not as weak as people think. Look at breaching charge for example, that can legit one shot a T3 behind a wall. Well placed air burst nade is also quite damaging. Main problem with shrapnel is that explosions tend to stun xenos, which makes shrapnel miss them completely.

its about the shrapnel accuracy and directional mechanics possibly being special/bugged in some way. The sharpnel we use for grenades is utterly ass because it misses too much (accuracy 8 vs accuracy 3 lmao), even if it didn't it's still spread in a circle so you're not risking such damage unless you're resting on top of the grenade. Funnily enough due to the dir stuff if a prae were to jump directly into the tile the OT claymore is placed on, it would get oneshot. Thankfully its pretty unlikely to happen
If im bored enough I might look into equalizing the accuracy numbers between shrapnel types since they're on two ridiculous sides of the spectrum.

Diegoflores31 commented:
> Shrapnel is not as weak as people think. Look at breaching charge for example, that can legit one shot a T3 behind a wall. Well placed air burst nade is also quite damaging. Main problem with shrapnel is that explosions tend to stun xenos, which makes shrapnel miss them completely.

you should know shrapnell is completely useless on grenades , as opposed to AGM or Bcharges , explosives launch their shraps in 9 directions . making them pretty much useless against xenomorphs + accuracy is horrible + damage is trash

Diegoflores31 commented:
from what i am aware ramdom number changes like this are usually on nono list for contributors  . 
the reasoning seems flawed and its just an overall nerf to all OT cases period

ihatethisengine commented:
> from what i am aware ramdom number changes like this are usually on nono list for contributors .
> the reasoning seems flawed and its just an overall nerf to all OT cases period

Cool, but what about the downsides?

Nanu308 commented:
Going to be looking into these two PR's this or next week, as i do agree OT rockets are too strong in particular. #7217 #7072

Diegoflores31 commented:
> Going to be looking into these two PR's this or next week, as i do agree OT rockets are too strong in particular. #7217 #7072

PR is not even tested 
> Played OT in the past, but didn't bother testing the new values so explosive falloff might require further tweaks to either side for desired intent.

GoldenDarkness55 commented:
> PR is not even tested

Doesn't really need to be, it's not going to break anything and if the problem persists you just further reduce values, simple enough.

Lukas-Seltz commented:
So my biggest problem here is that you did not test it. If you want to add something you need to test it that's just how it is.

Lukas-Seltz commented:
> > PR is not even tested
> 
> Doesn't really need to be, it's not going to break anything and if the problem persists you just further reduce values, simple enough.

No matter what is being changed you need to test it especially if it edits damage values and explosive values. OT requires skill just like CAS. If you have bad CAS they are useless if you have a Bad OT the hole lab blows up

Nanu308 commented:
> @Nanu308 I forget if tmerges can be messed with without breaking anything so imma hold on pushing commits, but while the rocket strength is fine now, I think the casing cost has to go up from 6 metal per rocket to 12 (more?) since just the fact OT can make 15-16 rockets off his roundstart reserves has a massive impact and ruins the ammo economy (3 rockets cost 3k while 50 metal for 8 OTs is 2k before cost increment)
> 
> Also range from 8 to either 7 (HE) or 6 (AP) since radius is around 3-4 tiles right now so FF isn't a concern. And possibly raise horrible accuracy (to AP rocket levels? inbetween?) since it's usually 1-2 tiles off mark, might be fine as a balance approach, but doesn't feel fair to screw RPG player over via RNG.
> 
> Lemme know which numbers to go with

Leave the rocket cost as is for now when it comes to metal. I am going to look at requisition as a whole and the metal economy at a later date. 

I'm fine with increasing the accuracy a lil bit, range can go down to 7-7.

GoldenDarkness55 commented:
> Leave the rocket cost as is for now when it comes to metal. I am going to look at requisition as a whole and the metal economy at a later date.
> 
> I'm fine with increasing the accuracy a lil bit, range can go down to 7-7.
> 
> Feel free to update PR with changes, I'll move it to live when it's done/am awake.

I really think the cost should go up regardless of future req/metal eco balance, 15 rockets off roundstart reserves could potentially end a game by themselves.

Also reverted rocket dmg since the buff wasn't needed and added slightly more falloff to grenades.

Diegoflores31 commented on code/datums/ammo/rocket.dm:
```diff
@@ -277,8 +277,9 @@
-	max_range = 8
+	accuracy = HIT_ACCURACY_TIER_5
+	accurate_range = 7
+	max_range = 7
```
wont this bring problems when shooting diagonally ??

GoldenDarkness55 replied:
no

ihatethisengine commented:
Dunno if you were trying to nerf OT mortar, but I just saw literally mini-OB with 3 screens stun diameter.

Nanu308 commented:
This has been fine in testing, further tweaks might be needed down the line, but a good step with bringing OT more in line regarding it's current power.

Lengord commented:
have you ever tried to play as OT?
   Its rare to get good OT by itself
   to stuff be used you have to make it by first drop, else half of the time it gets forgotten at FOB or Req, and marines already wiped by that time
   Pretty much all OT gamplay is to make some stuff from chems you have at start and then cryo, IF you want to torture yourself you can make flame nades or run around in hopes medical will make chems for you and they are mostly busy till 0:50, and it only enough for like five more nades or rockets, and so on. i mean you feel usefull as OT in about one out of five games
   people did rockets not because they were OP, but because they are one of only things that is usefull, have you ever seen someone making m40 explosive nade from starting chems?
Overall, my opinion OT was at edge of usefullness before, with all those changes it's becomming pointless play as OT

GoldenDarkness55 commented:
> Dunno if you were trying to nerf OT mortar, but I just saw literally mini-OB with 3 screens stun diameter.

Would have to see it in practice, mostly fine with mortars though. If it's tweaked in the future someone should just +10 the falloff again or just raise the minimum falloff from 25 to 35.

harryob reviewed (approved):
code
